src/t_cose_sign1_sign.c

```diff
@@ -45,10 +46,17 @@ t_cose_sign1_sign_init(struct t_cose_sign1_sign_ctx *me,
     } else
 #endif /* ! T_COSE_DISABLE_EDDSA */
     {
-        t_cose_signature_sign_main_init(&(me->signer.general),
-                                        me->cose_algorithm_id);
-        t_cose_sign_add_signer(&(me->me2),
-                      t_cose_signature_sign_from_main(&(me->signer.general)));
+        if(option_flags & T_COSE_OPT_RESTARTABLE) {
```

Comment:
@Was hoping all of the modifications to t_cose_sign1_sign could go away -- if you want to use restart you should just use t_cose_sign_sign. I think TFM will be better using t_cose_sign_sign when linking against t_cose 2.x -- code size will be smaller because the compatibility layer will be absent.  Maybe TFM will have to have an #ifdef T_COSE_2 

Additionally, and maybe more important, t_cose_sgin1_sign should not have a hard dependency on the restartable signer and thus on t_cose_crypto_sign_restart. 

The one and only way for a caller to request restart is t_cose_sign_add_signer(ctx, restartable_signer);